### PR TITLE
test(Button): Improve button tests

### PR DESCRIPTION
**Pull request by itwillwork**

### CODE REVIEW

File: playwright/core/expectScreenshotFixture.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const expectScreenshotFixture: PlaywrightFixture<ExpectScreenshotFixture>
     const expectScreenshot: ExpectScreenshotFixture = async ({
         component,
         screenshotName,
+        screenshotPostfix,
```
Comment (itwillwork) [inline playwright/core/expectScreenshotFixture.ts:13]: For multiple screenshots per test
`
 test(......... () => {
      await expectScreenshot();

      //...

      await expectScreenshot({
          screenshotPostfix: 'after hover',
      });
});
`

File: playwright/core/index.ts [hunk 2]
```diff
@@ -8,3 +8,5 @@ export const test = base.extend<Fixtures>({
     mount: mountFixture,
     expectScreenshot: expectScreenshotFixture,
 });
+
+export {expect} from '@playwright/experimental-ct-react';
```
Comment (itwillwork) [inline playwright/core/index.ts:12]: For import `expect` from `~playwright/core`

File: src/stories/tests-factory/create-smoke-scenarios.ts [hunk 3]
```diff
@@ -11,18 +12,20 @@ function checkIsCasesWithName<T>(cases: CasesWithName<T> | Cases<T>): cases is C
...
+    propsCases: Partial<{
+        [K in keyof Props]: CasesWithName<Props[K]> | Cases<Props[K]>;
+    }>,
     options?: Options,
 ) => {
     const scenarioDetails: ScenarioDetails = {
         tag: ['@smoke', ...(options?.additionalTags || [])],
     };
 
+    const scenarioName: ScenarioName = `smoke scenario${options?.scenarioName ? ` ${options?.scenarioName}` : ''}`;
```
Comment (itwillwork) [inline src/stories/tests-factory/create-smoke-scenarios.ts:24]: For a more declarative name of the generated test name

before

after

File: playwright/core/expectScreenshotFixture.ts (hunk 1 above)
Comment (amje) [reply playwright/core/expectScreenshotFixture.ts:13]: Maybe `screenshotNameSuffix`?

File: src/components/Button/__tests__/Button.visual.test.tsx [hunk 4]
```diff
@@ -89,20 +89,36 @@ test.describe('Button', {tag: '@Button'}, () => {
...
             await expectScreenshot();
+
+            if (props.disabled || props.loading) {
+                return;
+            }
+
+            await root.getByTestId(qa).hover();
+
+            await expectScreenshot({
+                screenshotPostfix: 'after hover',
```
Comment (amje) [inline src/components/Button/__tests__/Button.visual.test.tsx:120]: Let's use `hovered`, thus screenshot name will be more readable: `Button-smoke-hovered-...`

File: src/stories/tests-factory/create-smoke-scenarios.ts (hunk 3 above)
Comment (amje) [reply src/stories/tests-factory/create-smoke-scenarios.ts:24]: What if we shorten this to `smoke [default]` or `smoke [view: action]`, omit scenario and props words

File: playwright/core/expectScreenshotFixture.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ export const expectScreenshotFixture: PlaywrightFixture<ExpectScreenshotFixture>
     const expectScreenshot: ExpectScreenshotFixture = async ({
         component,
         screenshotName,
```
Comment (amje) [inline playwright/core/expectScreenshotFixture.ts:12]: Let's remove `screenshotName` (it's not used anywhere), and instead of `screenshotPostfix` introduce `nameSuffix` option

**Review by amje — CHANGES_REQUESTED** [review]

File: playwright/core/expectScreenshotFixture.ts (hunk 5 above)
Comment (itwillwork) [reply playwright/core/expectScreenshotFixture.ts:12]: Done

image

File: src/components/Button/__tests__/Button.visual.test.tsx (hunk 4 above)
Comment (itwillwork) [reply src/components/Button/__tests__/Button.visual.test.tsx]: renamed

**Review by amje — APPROVED** [review]
